opam package `coq.dev` broken in opam 1

<!-- Thank you for your contribution.
     Please complete the following information when reporting a bug. -->

#### Version

<!-- You can get this information by running `coqtop -v`. -->
master

#### Operating system
Linux

#### Description of the problem

`opam install coq.dev` is broken on our CI:
```
[coq: dune build] Command started
[ERROR] The compilation of coq failed at "dune build @vodeps".
[WARNING] Directory /builds/FP/LambdaRust-coq/opamroot/system/lib/coq is not
          empty, not removing

#=== ERROR while installing coq.dev ===========================================#
# opam-version         1.2.2
# os                   linux
# command              dune build @vodeps
# path                 /builds/FP/LambdaRust-coq/opamroot/system/build/coq.dev
# compiler             system (4.02.3)
# exit-code            1
# env-file             /builds/FP/LambdaRust-coq/opamroot/system/build/coq.dev/coq-2451-736596.env
# stdout-file          /builds/FP/LambdaRust-coq/opamroot/system/build/coq.dev/coq-2451-736596.out
# stderr-file          /builds/FP/LambdaRust-coq/opamroot/system/build/coq.dev/coq-2451-736596.err
### stderr ###
# => no CoqIde will be built.
# [...]
#   Bytecode VM enabled         : true
#   Native Compiler enabled     : true
# 
#   Local build, no installation...
# 
# If anything is wrong above, please restart './configure'.
# 
# *Warning* To compile the system for a new architecture
#           don't forget to do a 'make clean' before './configure'.
```
Seems like the transition to dune was not done properly in all places?

--- a/tools/coq_dune.ml
+++ b/tools/coq_dune.ml
@@ -80,6 +80,7 @@ module Aux = struct
   module DirMap = Map.Make(DirOrd)
 
   (* Functions available in newer OCaml versions *)
+  (* Taken from the OCaml std library (c) INRIA / LGPL-2.1 *)
   module Legacy = struct
 
     (* Slower version of DirMap.update, waiting for OCaml 4.06.0 *)
@@ -103,6 +104,29 @@ module Aux = struct
         end
       done;
       sub s 0 !j :: !r
+
+    (* Available in OCaml >= 4.04 *)
+    let is_dir_sep = match Sys.os_type with
+      | "Win32" -> fun s i -> s.[i] = '\\'
+      | _ -> fun s i -> s.[i] = '/'
+
+    let extension_len name =
+      let rec check i0 i =
+        if i < 0 || is_dir_sep name i then 0
+        else if name.[i] = '.' then check i0 (i - 1)
+        else String.length name - i0
+      in
+      let rec search_dot i =
+        if i < 0 || is_dir_sep name i then 0
+        else if name.[i] = '.' then check i (i - 1)
+        else search_dot (i - 1)
+      in
+      search_dot (String.length name - 1)
+
+    let remove_extension name =
+      let l = extension_len name in
+      if l = 0 then name else String.sub name 0 (String.length name - l)
+
   end
 
   let add_map_list key elem map =
@@ -181,18 +205,18 @@ let pp_vo_dep dir fmt vo =
   (* Correct path from global to local "theories/Init/Decimal.vo" -> "../../theories/Init/Decimal.vo" *)
   let deps = List.map (fun s -> sdir ^ s) (edep @ vo.deps) in
   (* The source file is also corrected as we will call coqtop from the top dir *)
-  let source = String.concat "/" dir ^ "/" ^ Filename.(remove_extension vo.target) ^ ".v" in
+  let source = String.concat "/" dir ^ "/" ^ Legacy.(remove_extension vo.target) ^ ".v" in
   (* The final build rule *)
   let action = sprintf "(chdir %%{project_root} (run coqtop -boot %s %s -compile %s))" eflag cflag source in
   pp_rule fmt [vo.target] deps action
 
 let pp_ml4_dep _dir fmt ml =
-  let target = Filename.(remove_extension ml) ^ ".ml" in
+  let target = Legacy.(remove_extension ml) ^ ".ml" in
   let ml4_rule = "(run coqp5 -loc loc -impl %{pp-file} -o %{targets})" in
   pp_rule fmt [target] [ml] ml4_rule
 
 let pp_mlg_dep _dir fmt ml =
-  let target = Filename.(remove_extension ml) ^ ".ml" in
+  let target = Legacy.(remove_extension ml) ^ ".ml" in
   let ml4_rule = "(run coqpp %{pp-file})" in
   pp_rule fmt [target] [ml] ml4_rule
 

